internal/osutils/autostart/autostart_darwin.go

```diff
@@ -0,0 +1,104 @@
+package autostart
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/ActiveState/cli/internal/assets"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/osutils/user"
+	"github.com/ActiveState/cli/internal/strutils"
+)
+
+const (
+	launchFileFormatName = "com.activestate.platform.%s.plist"
+	autostartFileSource  = "com.activestate.platform.autostart.plist.tpl"
+)
+
+func enable(exec string, opts Options) error {
+	enabled, err := isEnabled(exec, opts)
+	if err != nil {
+		return errs.Wrap(err, "Could not check if app autostart is enabled")
+	}
+
+	if enabled {
+		return nil
+	}
+
+	path, err := autostartPath(exec, opts)
+	if err != nil {
+		return errs.Wrap(err, "Could not get launch file")
+	}
+
+	installPath, err := installPath(opts.Name)
+	if err != nil {
+		return errs.Wrap(err, "Could not get install path")
+	}
+
+	asset, err := assets.ReadFileBytes(autostartFileSource)
+	if err != nil {
+		return errs.Wrap(err, "Could not read asset")
+	}
+
+	content, err := strutils.ParseTemplate(
+		string(asset),
+		map[string]interface{}{
+			"Label":       opts.MacLabel,
+			"Exec":        installPath,
+			"Interactive": opts.MacInteractive,
+		})
+	if err != nil {
+		return errs.Wrap(err, "Could not parse %s", fmt.Sprintf(launchFileFormatName, filepath.Base(exec)))
+	}
+
+	if err = fileutils.WriteFile(path, []byte(content)); err != nil {
+		return errs.Wrap(err, "Could not write launch file")
+	}
+	return nil
+}
+
+func disable(exec string, opts Options) error {
+	enabled, err := isEnabled(exec, opts)
+	if err != nil {
+		return errs.Wrap(err, "Could not check if app autostart is enabled")
+	}
+
+	if !enabled {
+		logging.Debug("Autostart is already disabled for %s", opts.Name)
+		return nil
+	}
+	path, err := autostartPath(exec, opts)
+	if err != nil {
+		return errs.Wrap(err, "Could not get launch file")
+	}
+	return os.Remove(path)
+}
+
+func isEnabled(exec string, opts Options) (bool, error) {
+	path, err := autostartPath(exec, opts)
+	if err != nil {
+		return false, errs.Wrap(err, "Could not get launch file")
+	}
+	return fileutils.FileExists(path), nil
+}
+
+func autostartPath(exec string, _ Options) (string, error) {
+	dir, err := user.HomeDir()
+	if err != nil {
+		return "", errs.Wrap(err, "Could not get home directory")
+	}
+	path := filepath.Join(dir, "Library/LaunchAgents", fmt.Sprintf(launchFileFormatName, filepath.Base(exec)))
+	return path, nil
+}
+
+func installPath(name string) (string, error) {
+	dir, err := user.HomeDir()
+	if err != nil {
+		return "", errs.Wrap(err, "Could not get home directory")
+	}
+	path := filepath.Join(dir, "Applications", fmt.Sprintf("%s.app", name))
+	return path, nil
+}
```

Comment:
This seems overly opinionated for autostart. We should just supply it the app path, it should not care about what it is.

Reply:
Updated this to use a function from the `installation` package and not assume the `Applications` directory. This change doesn't require us to update the autostart API